.github/workflows/lint.yml

```diff
@@ -116,16 +116,19 @@ jobs:
   security-lint:
     runs-on: ubuntu-latest
     steps:
-      - name: "Checkout"
+      - name: Checkout "security-lint" in-house CI action
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          # https://github.com/actions/checkout/issues/249
-          fetch-depth: 0
+          repository: instructlab/ci-actions
+          path: ci-actions
+          ref: v0.1.0
```

Comment:
We have two options for references... We can pin to an actual release, like this one (`v0.1.0`) or we can pin to a release branch like `release-v0.1` to allow for z-stream releases to be automatically consumed. (Note that `release-v0.1` is a protected branch and users are not allowed to directly commit to it.)

We cannot use logic liike:

```
ref: v0.1
```

to automatically pull in z-stream releases because `ref` only accepts the following inputs:
- branch name
- SHA
- GitHub tag

In order to reference a GitHub action by a major version (e.g., `@v3`) or a minor version (e.g., `@v3.2`), the action needs to be published to the GH marketplace.

Reply:
Seems the branch option will be more flexible and require less work on either in - I would prefer that

Reply:
+1 to the branch option given the alternatives